This patch fixes cweb.cls so that it works with new versions of the
LaTeX format.  (Any format later than 1997/06/30 or thereabouts is
affected.)

This patch may be freely redistributed.

To use it, run the command "patch -p0 < cweb.cls.patch" in the
directory containing cweb.cls if you have a UNIX-type system with the
patch command.  If not, just make the necessary changes by hand to
lines 846 and 848 as shown in the patch below.

   Julian Gilbey <jdg@debian.org>, November 1999

--- cweb.cls.orig
+++ cweb.cls
@@ -836,16 +836,16 @@
 
 %%% \sect Let's make sure that the expansion of |\end| is the one that we
 %%% expect. Then we can define |\cweb@skip_end| appropriately, we have to
-%%% skip 5~tokens.
+%%% skip 4~tokens.
 
 %%% \beginprog
 \CheckCommand*\end[1]{%
   \csname end#1\endcsname\@checkend{#1}%
   \expandafter\endgroup\if@endpe\@doendpe\fi
-% 1         2      3            4            5
-  \if@ignore\global\@ignorefalse\ignorespaces\fi}
+% 1         2            3            4
+  \if@ignore\@ignorefalse\ignorespaces\fi}
 
-\def\cweb@skip_end#1#2#3#4#5{\cweb@check_fi}
+\def\cweb@skip_end#1#2#3#4{\cweb@check_fi}
 %%% \endprog
 
 
